storage/sis-storage/src/main/java/org/apache/sis/storage/tilematrix/TileMatrix.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.sis.storage.tilematrix;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.apache.sis.coverage.grid.GridExtent;
+import org.apache.sis.coverage.grid.GridGeometry;
+import org.apache.sis.storage.DataStoreException;
+import org.opengis.util.GenericName;
+
+/**
+ * A TileMatrix is collection of tiles with the same size and properties placed
+ * on a regular grid with no overlaping.
+ * While a TileMatrix in a regular 2D grid space it may still be defined on other dimensions,
+ * it's GridGeometry may have any number of additional axes.
+ *
+ * @author Johann Sorel (Geomatys)
+ */
+public interface TileMatrix {
+
+    /**
+     * @return unique identifier.
+     */
+    GenericName getIdentifier();
+
+    /**
+     * Returns the structure of the tile matrix.
+     * Returned {@linkplain GridGeometry} is always a two dimensional slice, it may still contain other dimension such as time and
+     * elevation.
+     * Returned {@linkplain GridGeometry} has a linear grid to crs transform which in most cases do not have rotation or shearing.
+     *
+     * @return {@linkplain GridGeometry} never null.
+     */
+    GridGeometry getGeometry();
+
+    /**
+     * Request a tile status.
+     * The accuracy of a tile status greatly varies with each protocol.
+     *
+     * @param col tile column index
+     * @param row tile row index
+     * @return tile status, never null
+     * @throws DataStoreException if fetching the tile status failed.
+     */
+    TileStatus getTileStatus(long col, long row) throws DataStoreException;
+
+    /**
+     * Find a {@linkplain Tile} in the TileMatrix.
+     * This tile should be fast to access, it's purpose is to provide a sample
+     * of content of the TileMatrix.
+     *
+     * The returned tile may not really exist or be part of this TileMatrix at all.
+     *
+     * @return tile, should not be null.
+     * @throws DataStoreException if fetching the tile failed.
+     */
+    Tile anyTile() throws DataStoreException;
```

Comment:
Maybe we should omit this method, or hide it in some internal API until we have verified if it is really needed. It the intend is to get any tile, the same functionality could be achieved with `getTiles(…).findAny()`. It the intend is to get a template, we may rename this method as something like `template()` or investigate for ways to provide the desired metadata without an "read any (pseudo) tile" method.